gno.land/pkg/gnoland/app_test.go

```diff
@@ -1,20 +1,173 @@
 	"github.com/gnolang/gno/tm2/pkg/events"
+	"github.com/gnolang/gno/tm2/pkg/log"
 	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/gnolang/gno/tm2/pkg/store"
+	"github.com/gnolang/gno/tm2/pkg/store/dbadapter"
+	"github.com/gnolang/gno/tm2/pkg/store/iavl"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+// Tests that NewAppWithOptions works even when only providing a simple DB.
+func TestNewAppWithOptions(t *testing.T) {
+	app, err := NewAppWithOptions(&AppOptions{
+		DB: memdb.NewMemDB(),
+		InitChainerConfig: InitChainerConfig{
+			CacheStdlibLoad:        true,
+			GenesisTxResultHandler: PanicOnFailingTxResultHandler,
+			StdlibDir:              filepath.Join(gnoenv.RootDir(), "gnovm", "stdlibs"),
+		},
+	})
+	require.NoError(t, err)
+	bapp := app.(*sdk.BaseApp)
+	assert.Equal(t, "dev", bapp.AppVersion())
+	assert.Equal(t, "gnoland", bapp.Name())
+
+	addr := crypto.AddressFromPreimage([]byte("test1"))
+	resp := bapp.InitChain(abci.RequestInitChain{
+		Time:    time.Now(),
+		ChainID: "dev",
+		ConsensusParams: &abci.ConsensusParams{
+			Block: &abci.BlockParams{
+				MaxTxBytes:   1_000_000,   // 1MB,
+				MaxDataBytes: 2_000_000,   // 2MB,
+				MaxGas:       100_000_000, // 100M gas
+				TimeIotaMS:   100,         // 100ms
+			},
+		},
+		Validators: []abci.ValidatorUpdate{},
+		AppState: GnoGenesisState{
+			Balances: []Balance{
+				{
+					Address: addr,
+					Amount:  []std.Coin{{Amount: 1e15, Denom: "ugnot"}},
+				},
+			},
+			Txs: []std.Tx{
+				{
+					Msgs: []std.Msg{vm.NewMsgAddPackage(addr, "gno.land/r/demo", []*std.MemFile{
+						{
+							Name: "demo.gno",
+							Body: "package demo; func Hello() string { return `hello`; }",
+						},
+					})},
+					Fee:        std.Fee{GasWanted: 1e6, GasFee: std.Coin{Amount: 1e6, Denom: "ugnot"}},
+					Signatures: []std.Signature{{}}, // one empty signature
+				},
+			},
+		},
+	})
+
+	if !resp.IsOK() {
+		t.Fatal(resp)
+	}
+
+	tx := amino.MustMarshal(std.Tx{
+		Msgs: []std.Msg{vm.NewMsgCall(addr, nil, "gno.land/r/demo", "Hello", nil)},
+		Fee: std.Fee{
+			GasWanted: 100_000,
+			GasFee: std.Coin{
+				Denom:  "ugnot",
+				Amount: 1_000_000,
+			},
+		},
+		Signatures: []std.Signature{{}}, // one empty signature
+		Memo:       "",
+	})
+	dtxResp := bapp.DeliverTx(abci.RequestDeliverTx{
+		RequestBase: abci.RequestBase{},
+		Tx:          tx,
+	})
+	if !dtxResp.IsOK() {
+		t.Fatal(dtxResp)
+	}
+}
+
+func TestNewAppWithOptions_ErrNoDB(t *testing.T) {
+	_, err := NewAppWithOptions(&AppOptions{})
+	assert.ErrorContains(t, err, "no db provided")
+}
+
+// Test whether InitChainer calls to load the stdlibs correctly.
+func TestInitChainer_LoadStdlib(t *testing.T) {
+	t.Run("cached", func(t *testing.T) { testInitChainerLoadStdlib(t, true) })
+	t.Run("uncached", func(t *testing.T) { testInitChainerLoadStdlib(t, false) })
+}
+
+func testInitChainerLoadStdlib(t *testing.T, cached bool) { //nolint:thelper
```

Comment:
Why nolint?

Reply:
You use `t` for assertions in this function

Reply:
```
func (c *T) Helper()
    Helper marks the calling function as a test helper function. When printing
    file and line information, that function will be skipped. Helper may be
    called simultaneously from multiple goroutines.
```

The test here _is_ properly a test, and I want the line reported on the log/error to be the line of the assertion, rather than line 113 (where `testInitChainerLoadStdlib` is called).